modules/sandboxed-containers-preparing-openshift-cluster.adoc

```diff
@@ -3,5 +3,96 @@
+
+[id="sandboxed-containers-additional-resource-requirements_{context}"]
+== Additional resource requirements for {sandboxed-containers-first}
+
+{sandboxed-containers-first} is a product that brings the ability to run workloads inside a sandboxed runtime, such as Kata Containers, to your {product-title} clusters. Each pod is represented by a virtual machine (VM). Each VM hosts a `kata-agent` that processes guests as a supervisor for managing containers and running your container workloads. There are two additional processes that add more overhead:
+
+* `containerd-shim-kata-v2` is used to communicate with the pod.
+* `Virtiofsd` handles host file system access on behalf of the guest.
+
+Each VM is configured with a default amount of memory. Additional memory is hot-plugged into the VM for containers that explicitly request memory.
+
+* If a container runs without a given memory resource, it is able to consume free memory. It will do so until the total memory used by the VM reaches the default allocation. The guest and its I/O buffers also consume memory.
+* If a container is given a specific amount of memory, then that memory is hot-plugged into the VM before the container starts.
+* If a memory limit is specified, then the memory limit is enforced within the VM using cgroups. If the container exceeds the specified limit, it is OOM-killed.
+
+.Indicative memory sizes
+
+The table below lists some typical values for resource allocation.
+
+[cols="2,2"]
+|===
+|Resource |Value
+
+|Memory allocated by default to a virtual machine | 2G
+|Guest Linux kernel memory usage at boot | ~110M
+|Memory used by the qemu process (excluding VM memory) | ~30M
+|Memory used by the virtiofsd process (excluding VM I/O buffers) | ~10M
+|Memory used by the shimv2 process | ~20M
+|File buffer cache data after a ‘dnf install’ on Fedora | ~300M* ^[1]^
+|===
+[.small]
+--
+1. File buffers show up and are accounted for in multiple locations:
+* In the guest as guest file file buffer cache.
+* In the `virtiofsd` daemon mapped in order to allow I/O.
+* In the QEMU process as guest memory.
+--
+
+This is properly accounted for by the memory utilization metrics, which only count that memory once.
+
+This is described by a link:https://kubernetes.io/docs/concepts/scheduling-eviction/pod-overhead/[pod overhead]. You can get the current pod overhead for the `kata` runtime class by using `oc describe runtimeclass kata` as shown below.
+
+.Example
+[source,terminal]
+----
+$ oc describe runtimeclass kata
+Name:         kata
+[...]
+Metadata:
+[...]
+Overhead:
+  Pod Fixed:
+    Cpu:     250m
+    Memory:  350Mi
+[...]
+----
+
+You can change the pod overhead by changing the `spec.overhead` field for a `RuntimeClass`. For instance, if the configuration that you run for your containers consumes more than 350Mi of memory for the QEMU process and guest kernel data, you may alter the `RuntimeClass` overhead to suit your needs.
```

Comment:
Here we wanted to add a note saying something like: "The default overhead values have been tested by RH, please only change them if you are aware of possible consequences"

Reply:
Christophe mentioned he will be adding it, problem is, I can't tag him anymore not sure why 🤔

Reply:
@sagidlow I like the additional sentence @jensfr suggested. I'm still awaiting my openshift org inclusion, the email they sent me went to spam.